### PR TITLE
Make repositories created with new Chorus load in old versions

### PR DESCRIPTION
**Pull request by jasonleenaylor**
* Move the allownumberbranch into the fixutf8 extensions.
  (adding a new extension breaks backward compatibility)
* Adjust installer source
* Remove the allownumberbranch as an extension

### CODE REVIEW

**Comment by jasonleenaylor** [comment]
CI builds for this change are currently succeeding:
http://build.palaso.org/viewType.html?buildTypeId=Chorus_NewHgLinux
http://build.palaso.org/viewType.html?buildTypeId=Chorus_win32_master_NewHg

File: src/LibChorus/VcsDrivers/Mercurial/HgRepository.cs [hunk 1]
```diff
@@ -174,8 +174,8 @@ internal void CheckAndUpdateHgrc()
 			{
 				/*
 					fixutf8 makes it possible to have unicode characters in path names. Note that it is prone to break with new versions of mercurial.
-					it works with 1.5.1, and reportedly with 1.84, but the version I got did not work with 1.9.2.
-					When updating, notice that there are several forks available
+					We currently have one tweaked to work with all tested versions of Mercurial 3
+					When updating you will likely have to modify our version since people who need this us git now.
```
Comment (JohnThomson) [inline src/LibChorus/VcsDrivers/Mercurial/HgRepository.cs]: I think they really USE git

File: MercurialExtensions/fixutf8/.guidsForInstaller.xml [hunk 2]
```diff
@@ -15,6 +15,8 @@
   <File Id="win32helper.py" Guid="1bfc3f08-eed2-4c80-9ae6-2cf6993adaf2" />
   <File Id="win32helper.pyc" Guid="0fb5d787-016d-4fa0-b68f-9e117606cab6" />
   <File Id="win32helper.pyo" Guid="20da0825-1a7f-4948-a1b7-70f29d8d0d16" />
+  <File Id="MercurialExtensions.fixutf8.allownumberbranch.py" Guid="683506C7-4607-489A-A9F6-215542560276"/>
+  <File Id="MercurialExtensions.fixutf8.allownumberbranch.pyc" Guid="2D8D8D79-DFFE-4CFE-91C4-CA0EF843A2BF"/>
```
Comment (JohnThomson) [inline MercurialExtensions/fixutf8/.guidsForInstaller.xml]: We were supposed to be getting rid of this, not adding it

File: MercurialExtensions/fixutf8/fixutf8.py [hunk 3]
```diff
@@ -43,79 +43,108 @@
...
+# extsetup is only required on windows
+def extsetup():
+	if sys.platform == 'win32':
+		winextsetup()
 
-stdout = sys.stdout
+# uisetup on linux needs only to enable number only branches
+def uisetup(ui):
+	allownumberbranches_uisetup()
+	if sys.platform == 'win32':
```
Comment (JohnThomson) [inline MercurialExtensions/fixutf8/fixutf8.py]: Comment explaining this was the original whole thing

**Comment by jasonleenaylor** [comment]
Adjusted PR as result of JT review

**Pull request merged**